    Repository Purpose

This repository contains the files I created during the dissertation project on my Master's degree in Bioinformatics at Newcastle University. This project took place between April and August 2021. The R and Bash scripts I created during my data analysis as well as the 'README' file for the project can be found within. In addition to these scripts I also used software developed by other scientists for my project and the links to them can be found at the end of this description.

This was my first long term project in bioinformatics so I apologise in advance for any mistakes regarding good coding/bioinformatics practice. I hope to improve and correct my practice as I advance throughout my career and believe keeping a record of this amateur work is useful for purposes of self reflection.

    Dissertation Project

The thesis project for my master's degree was titled: 'Investigation of the genetic drivers of cell differentiation in Human Haematopoietic Foetal Liver Stem Cells using Population Balance Analysis.'

In brief the aim of my project was to asses the usefulness of the Population Balance Analysis (PBA) algorithm presented in the paper 'Fundamental limits on dynamic inference from single-cell snapshots' (Weinreb C, et al, 2018, DOI: 10.1073/pnas.1714723115) in determinng cell fate and ordering cells as they progress down set lineages. To do so a large collection of single-cell RNA sequencing data from a recent paper 'Decoding human fetal liver haematopoiesisâ€™ (Popescu D.M., et al, 2019, DOI: 10.1038/s41586-019-1652-y) was chosen to test the PBA algorithm with a larger dataset and to gauge its usefulness at ordering cells with an already determined cell order. This data included gene expression profiles for haematopoeitc stem cells (HSC) and the cell states that make up the possible mature cell states it can develop towards such as: B-cells, T-cells, Erythrocytes, Monocyte, Megacaryocytes to list but a few. The difference in gene expression between cells chosing different paths at bifurcations was also to be analysed using data as such analysis would help determine at which point a cell is locked into or has a bias towards a particular lineage over another and how the gene expression profile changes surrounding this.

Unfortunately the Covid-19 pandemic began and the UK went into lockdown and the project had to be completed away from the university. Due to this the project had to alter to suit the available hardware on my PC (The main botrtleneck relating to available RAM). This caused the project to shift focus towards only looking at how the PBA algorithm ordered cells developing down one lineage and how the ordering compared to how the cells were ordered as displayed in the Human Cell Atlas which acompanied the paper by Popescu D.M., et al. This comparison was made by comparing expression pattern of eight genes progressing from HSC down the B-cell lineage (from HSC through to mature B cells) as ordered by the PBA algorithm to that displayed in the paper the data was from. 

The results showed indicated that general gene expression pattern was similar for some of the genes but not for others and the ordering was rather noisy and required some manual ordering/correction. However, this may have been due to the settings of the PBA not being perfect which resulted in the knn-graph it produced having a mosaic pattern rather than the expected smooth transition progressing down the lineage. Due to time constraints I was unable to optomise the settings to attempt to correct this as the analysis required the use of a shared HPC to run the PBA algorithm and I didnt want to disturb other student's or researcher's work. This is one aspect I would like to check out should I have the time and hardware access required. In addition I would enjoy the oppertunity to test the PBA algorithm over different lineages and to include bifurcations to judge its ability at determining cell fate which was the intial aim of the project.
